### ISSUE TITLE
delete_metadata frees metadata a bit too early

### ISSUE THREAD

**Issue by nritsche** [kotekan/kotekan#516]
branch `develop`:  `delete_metadata()` in `lib/core/metadata.c` tries to access `container` just after it frees it:

```
void delete_metadata(struct metadataContainer * container) {
    ...
    free(container);
    CHECK_ERROR_F( pthread_mutex_destroy(&container->metadata_lock) );
}
```

**Comment by andrerenard** [comment]
Looks like this was a regression that showed up in #512  Sorry I missed that in the review.

**Comment by james-s-willis** [comment]
That's my bad. I've fixed it in #518.
